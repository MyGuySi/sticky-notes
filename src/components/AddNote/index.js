// @flow

import React from 'react';
import KeyboardShortcut from '../KeyboardShortcut';
import Button from '../Button';
import styles from './styles.module.css'

type Props = {
  isNotesEmpty: bool,
  onClickAdd: () => void,
}

function AddNote (props: Props) {
  return (
    <div className={styles.wrapper}>
      <KeyboardShortcut value='n' handler={props.onClickAdd} blockEvent />
      <Button onClick={props.onClickAdd} data-testid='add-button'>New note (n)</Button>
      {props.isNotesEmpty && (
        <img className={styles.empty} src={require('../../images/get-started.svg')} width='225' height='83' alt='' />
      )}
    </div>
  )
}

export default AddNote