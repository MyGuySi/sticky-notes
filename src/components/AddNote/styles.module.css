.wrapper {
  position: relative;
}

.wrapper > * {
  width: 100%;
  height: 100%;
}

.empty {
  width: 14.5rem;
  height: 4.1rem;
  position: absolute;
  right: -12rem;
  bottom: -5rem;
}

@media screen and (max-width: calc(600px + 5rem - 1px)) {
  .empty {
    right: 0;
    bottom: -6rem;
  }
}