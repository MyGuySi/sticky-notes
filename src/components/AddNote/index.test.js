// @flow

import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AddNote from './index'

it('renders correctly', () => {
  const component = renderer.create(
    <AddNote isNotesEmpty={true} onClickAdd={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls add handler', () => {
  const handler = jest.fn();
  render(
    <AddNote isNotesEmpty={true} onClickAdd={handler} />
  )
  fireEvent.click(screen.getByTestId('add-button'));
  expect(handler.mock.calls.length).toBe(1);
});