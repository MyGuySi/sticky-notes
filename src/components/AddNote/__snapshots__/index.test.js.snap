// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <button
    className="button"
    data-testid="add-button"
    onClick={[Function]}
  >
    New note (n)
  </button>
  <img
    alt=""
    className="empty"
    height="83"
    src={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "get-started.svg",
      }
    }
    width="225"
  />
</div>
`;
